bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala

```diff
@@ -0,0 +1,46 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.spModel.config2.{Config, ConfigSequence, ItemKey}
+import edu.gemini.spModel.obscomp.InstConstants.INSTRUMENT_NAME_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Created by jluhrs on 4/28/15.
+ */
+sealed class Executor(sequenceConfig: ConfigSequence) {
+
+  def run = runSequence(sequenceConfig.getAllSteps.toList)
+
+  private def runSequence(config: List[Config]): Future[Int] = {
+    def chainStep(config: List[Config], prevFuture: Future[Int]): Future[Int] = config match {
+      case List() => prevFuture
+      case x :: ys => chainStep(ys, prevFuture flatMap (i => runStep(x, i)))
+    }
+    config match {
+      case List() => Promise.failed(new Exception("Empty sequence configuration")).future
+      case x :: ys => chainStep(ys, runStep(x, 0))
+    }
+  }
```

Comment:
Just a few stylistic things:
- Use `Nil` when matching on an empty list.
- Use `Future.failed(ex)` rather than `Promise` if you want to construct a failed future.
-  This method can be written as a fold, something like this:

```scala
private def runSequence(config: List[Config]): Future[Int] =
  config match {
    case Nil => Future.failed(new Exception("Empty sequence configuration")) // but is this really an error?
    case _   => config.foldLeft(Future(0))((f, s) => f.flatMap(runStep(s, _)))
  }
```